lib/spack/spack/spec.py

```diff
@@ -1554,6 +1555,29 @@ def spliced(self):
         """
         return any(s.build_spec is not s for s in self.traverse(root=True))
 
+    @property
+    def installed(self):
+        """Installation status of a package.
+
+        Returns:
+            True if the package has been installed, False otherwise.
+        """
+        try:
```

Comment:
These methods should either assert that the spec isn’t concrete or just return false if it’s concrete.

I can’t decide which is better. I lean towards asserting since I think it’s probably an error if you ask an abstract spec whether it’s installed, but I could see returning False as maybe a less prickly API. I still kind of think something’s wrong with the client code if it has to ask this of an abstract spec.

Reply:
Fixed in f862dce